File: Random Codes/sieve.cpp
#include <iostream>
#include <vector>
using namespace std;

const int n = 10000000;

int main() {

    vector<bool> primes(n + 1, true);
    vector<int> primeNumbers;

    primes[0] = primes[1] = false;

    for (int p = 2; p * p <= n; ++p) {
        if (primes[p] == true) {
            for (int i = p * p; i <= n; i += p) {
                primes[i] = false;
            }
        }
    }

    for (int i = 2; i <= n; ++i) {
        if (primes[i]) {
            primeNumbers.push_back(i);
        }
    }

    int num;
    cout<<"Enter Number\n";
    cin>>num;

    if(primes[num]) cout<<"Prime Number\n";
    else cout<<"Composite Number\n";

    return 0;
}

File: Random Codes/sieve.py
n = 10000000

primes = [True] * (n + 1)
prime_numbers = []

primes[0] = primes[1] = False

for p in range(2, int(n**0.5) + 1):
    if primes[p]:
        for i in range(p * p, n + 1, p):
            primes[i] = False

for i in range(2, n + 1):
    if primes[i]:
        prime_numbers.append(i)

num = int(input("Enter Number:\n"))

if primes[num]:
    print("Prime Number")
else:
    print("Composite Number")
